### PR TITLE
chore: add serialization test from python

### PR DESCRIPTION
**Pull request by nimrod-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)

### CODE REVIEW

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: 0 of 4 files reviewed, 1 unresolved discussion

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/69#-Nwi1Ns-2q5yNY7I7QxG:-Nwi1Ns-2q5yNY7I7QxH:brpxi3w) (no related file):*
[python side PR](https://reviewable.io/reviews/starkware-industries/starkware/34726)

___

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 5 unresolved discussions (waiting on @nimrod-starkware)

___
*[`crates/committer/src/deserialization/read.rs` line 25 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiBrjn3miC-ezqfuCo:-NwiBrjn3miC-ezqfuCp:berryzn) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L25)):*
> ```XML
>         Err(_) => "Test failed :(\n".to_string(),
>     }
> }
> ```

1. `res` is a redundant variable
2. this function should output a "python test result" IMO
3. since `PythonTestError` has a variant called `ParseInputError`, consider using it
4. on key-duplicate error - you can panic, or add a variant in the python test error enum... your call



_Suggestion:_
```XML
pub fn python_input_parsing_test(temp_input_path: String) -> Result<String, PythonTestError> {
    let input = fs::read_to_string(temp_input_path).expect("Unable to read given file.");
    match parse_input(input) {
        Ok(actual_input) => Ok(create_output_to_python(actual_input)),
        Err(DeserializationError::ParsingError(serde_error)) => Err(PythonTestError::from(serde_error)),
        Err(DeserializationError::KeyDuplicate(key)) => panic!("Duplicate key in input test: {key}."),
    }
}
```
___
*[`crates/committer/src/deserialization/read.rs` line 50 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiAVWs8YSC6QZk_fxK:-NwiAVWs8YSC6QZk_fxL:byowhpx) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L50)):*
> ```XML
>         actual_input.tree_height.0
>     )
> }
> ```

python-related code should not be in the `committer` crate; move to `committer_cli` crate.



_Code quote:_
```XML
pub fn python_input_parsing_test(temp_input_path: String) -> String {
    let input = fs::read_to_string(temp_input_path).expect("Unable to read given file.");
    let res = parse_input(input);
    match res {
        Ok(actual_input) => create_output_to_python(actual_input),
        Err(_) => "Test failed :(\n".to_string(),
    }
}

fn create_output_to_python(actual_input: Input) -> String {
    format!(
        r#"
        {{
        "storage_size": {},
        "address_to_class_hash_size": {},
        "address_to_nonce_size": {},
        "class_hash_to_compiled_class_hash": {},
        "current_contract_state_leaves_size": {},
        "outer_storage_updates_size": {},
        "tree_height": {}
        }}"#,
        actual_input.storage.len(),
        actual_input.state_diff.address_to_class_hash.len(),
        actual_input.state_diff.address_to_nonce.len(),
        actual_input
            .state_diff
            .class_hash_to_compiled_class_hash
            .len(),
        actual_input.state_diff.current_contract_state_leaves.len(),
        actual_input.state_diff.storage_updates.len(),
        actual_input.tree_height.0
    )
}
```
___
*[`crates/committer/src/deserialization/read.rs` line 50 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiB8Mk3SYiDy6_eBo6:-NwiB8Mk3SYiDy6_eBo7:b-dj2ca9) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L50)):*
> ```XML
>         actual_input.tree_height.0
>     )
> }
> ```

this test only tests that the *amount* of deserialized arguments are correct and that the general format is ok;
I am also interesting to see if the values themselves are somehow mangled.

To make things simple, take all the values and hash them in some deterministic way, and then output the final hash.
compute the same hash on the python side and compare (seems easier than comparing values one by one)



_Code quote:_
```XML
fn create_output_to_python(actual_input: Input) -> String {
    format!(
        r#"
        {{
        "storage_size": {},
        "address_to_class_hash_size": {},
        "address_to_nonce_size": {},
        "class_hash_to_compiled_class_hash": {},
        "current_contract_state_leaves_size": {},
        "outer_storage_updates_size": {},
        "tree_height": {}
        }}"#,
        actual_input.storage.len(),
        actual_input.state_diff.address_to_class_hash.len(),
        actual_input.state_diff.address_to_nonce.len(),
        actual_input
            .state_diff
            .class_hash_to_compiled_class_hash
            .len(),
        actual_input.state_diff.current_contract_state_leaves.len(),
        actual_input.state_diff.storage_updates.len(),
        actual_input.tree_height.0
    )
}
```
___
*[`crates/committer_cli/src/tests/python_tests.rs` line 41 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiD4A15sFSvtqMcLms:-NwiD4A15sFSvtqMcLmt:buredio) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer_cli/src/tests/python_tests.rs#L41)):*
> ```XML
>                 Ok(example_test(example_input))
>             }
>             Self::InputParsing => Ok(parse_input_test(input)),
> ```

see above (parse_input_test should return a Result)



_Suggestion:_
```XML
parse_input_test(input)
```

**Review by dorimedini-starkware — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 4 unresolved discussions (waiting on @nimrod-starkware)

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 4 unresolved discussions (waiting on @dorimedini-starkware)

___
*[`crates/committer/src/deserialization/read.rs` line 25 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiBrjn3miC-ezqfuCo:-NwiRv39E_BH92PoRR6-:b-zfk2mg) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L25)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

1. `res` is a redundant variable
2. this function should output a "python test result" IMO
3. since `PythonTestError` has a variant called `ParseInputError`, consider using it
4. on key-duplicate error - you can panic, or add a variant in the python test error enum... your call
</blockquote></details>

1. agree
about 2,3,4 IIUC, `PythonTestError` indicates that it couldn't run the given test. question is, how do we want to handle a situation where the test could run but failed? i think there are two options -
1. print an indicative message and catch it in python.
2. panic at rust with some indicative message
___
*[`crates/committer/src/deserialization/read.rs` line 50 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiAVWs8YSC6QZk_fxK:-NwiUlZX5274wth95Td4:b6cpen6) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L50)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

python-related code should not be in the `committer` crate; move to `committer_cli` crate.
</blockquote></details>

that's what i wanted to do in the first place. 
The problem is I don't want to externalize the `Input` and `DeserializationResult` structs to the `committer_cli` crate. (it requires changing all inner structs from `pub(crate)` to `pub` and I don't think it's a good way to solve it).

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: 3 of 4 files reviewed, 4 unresolved discussions (waiting on @dorimedini-starkware)

___
*[`crates/committer/src/deserialization/read.rs` line 50 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiAVWs8YSC6QZk_fxK:-NwiZ9kh-Z97JvJ_AJG7:b-ugi2v6) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L50)):*
<details><summary><i>Previously, nimrod-starkware wrote…</i></summary><blockquote>

that's what i wanted to do in the first place. 
The problem is I don't want to externalize the `Input` and `DeserializationResult` structs to the `committer_cli` crate. (it requires changing all inner structs from `pub(crate)` to `pub` and I don't think it's a good way to solve it).
</blockquote></details>

maybe we should move the whole `deserialization` directory to `committer_cli`?

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 5 unresolved discussions (waiting on @nimrod-starkware)

___
*[`crates/committer/src/deserialization/read.rs` line 25 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiBrjn3miC-ezqfuCo:-NwicULU5mKfQgsqb2by:b-hkrrcd) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L25)):*
<details><summary><i>Previously, nimrod-starkware wrote…</i></summary><blockquote>

1. agree
about 2,3,4 IIUC, `PythonTestError` indicates that it couldn't run the given test. question is, how do we want to handle a situation where the test could run but failed? i think there are two options -
1. print an indicative message and catch it in python.
2. panic at rust with some indicative message
</blockquote></details>

I like error objects because they contain the "stack trace".
I would suggest we do the following:
1. Define a top-level error in the committer crate - say, `CommitterError`.
2. Add a `Committer(#[from] CommitterError)` in the `PythonTestError` enum

This way, python tests can run any committer code and wrap errors transparently.
When running a test from python side, if it fails, by printing out the error object itself we will be able to see what happened.

tl;dr:
1. Tests should return PythonTestError objects (they are allowed to panic if it makes sense to panic, but errors are preferable).
2. In the CLI's `main` function, after running a python test, convert errors to panics.
___
*[`crates/committer/src/deserialization/read.rs` line 50 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiAVWs8YSC6QZk_fxK:-NwidjsgBG03wkVYHgxo:b-hkjyn6) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L50)):*
<details><summary><i>Previously, nimrod-starkware wrote…</i></summary><blockquote>

maybe we should move the whole `deserialization` directory to `committer_cli`?
</blockquote></details>

only the from-python-to-storage logic should be in the CLI crate.
from-storage-to-original-skeleton needs to be in the committer crate.

In any case, we *cannot* refer to python in the committer crate, so if there is no other solution (like moving deserialization logic) you will need to change lots of `pub(crate)`s to `pub`s...
___
*[`crates/committer/src/deserialization/read.rs` line 24 at r3](https://reviewable.io/reviews/starkware-libs/committer/69#-NwideV9-QSdm_FtyVe3:-NwideVA2TAHZUc_6f6O:b59k9e9) ([raw file](https://github.com/starkware-libs/committer/blob/b6864072fb31accbf6ecdafe98a96fe77f085cbe/crates/committer/src/deserialization/read.rs#L24)):*
> ```XML
>         Err(DeserializationError::ParsingError(serde_error)) => {
>             format!("Test failed. {}", serde_error)
>         }
> ```

Prefer error objects



_Code quote:_
```XML
        Err(DeserializationError::KeyDuplicate(key)) => format!("Test failed. {}", key),
        Err(DeserializationError::ParsingError(serde_error)) => {
            format!("Test failed. {}", serde_error)
        }
```

**Review by dorimedini-starkware — COMMENTED** [review]
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 5 unresolved discussions (waiting on @nimrod-starkware)

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: 3 of 4 files reviewed, 5 unresolved discussions (waiting on @dorimedini-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 41 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiD4A15sFSvtqMcLms:-Nwj2qm-6_aEl0XFn8Jf:b-896fix) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer_cli/src/tests/python_tests.rs#L41)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

see above (parse_input_test should return a Result)
</blockquote></details>

Done.
___
*[`crates/committer/src/deserialization/read.rs` line 25 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiBrjn3miC-ezqfuCo:-Nwj4E1B0fU_3oHjhTSb:bijex0d) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L25)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

I like error objects because they contain the "stack trace".
I would suggest we do the following:
1. Define a top-level error in the committer crate - say, `CommitterError`.
2. Add a `Committer(#[from] CommitterError)` in the `PythonTestError` enum

This way, python tests can run any committer code and wrap errors transparently.
When running a test from python side, if it fails, by printing out the error object itself we will be able to see what happened.

tl;dr:
1. Tests should return PythonTestError objects (they are allowed to panic if it makes sense to panic, but errors are preferable).
2. In the CLI's `main` function, after running a python test, convert errors to panics.
</blockquote></details>

I added `TestFailure` entry to `PythonTestError` which can be casted from `DeserializationError`. WDYT?
___
*[`crates/committer/src/deserialization/read.rs` line 24 at r3](https://reviewable.io/reviews/starkware-libs/committer/69#-NwideV9-QSdm_FtyVe3:-Nwj68Lo1urPL-dvZn4j:b-896fix) ([raw file](https://github.com/starkware-libs/committer/blob/b6864072fb31accbf6ecdafe98a96fe77f085cbe/crates/committer/src/deserialization/read.rs#L24)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

Prefer error objects
</blockquote></details>

Done.

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 3 unresolved discussions (waiting on @nimrod-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 62 at r5](https://reviewable.io/reviews/starkware-libs/committer/69#-Nwj6YQc6gBjuVBFYXpc:-Nwj6YQc6gBjuVBFYXpd:b7fpkrj) ([raw file](https://github.com/starkware-libs/committer/blob/db15b6d5e6f47df044d8e83d1562d388f488dfb4/crates/committer_cli/src/tests/python_tests.rs#L62)):*
> ```XML
>         Ok(actual_input) => Ok(create_output_to_python(actual_input)),
>         Err(err) => Err(PythonTestError::from(err)),
>     }
> ```

no more need for the `match`; the `?` operator automatically does `PythonTestError::from`



_Suggestion:_
```XML
    Ok(create_output_to_python(parse_input(input)?))
```
___
*[`crates/committer/src/deserialization/read.rs` line 25 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiBrjn3miC-ezqfuCo:-Nwj6CCH0irnCzmDCNU_:b3bidlg) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L25)):*
<details><summary><i>Previously, nimrod-starkware wrote…</i></summary><blockquote>

I added `TestFailure` entry to `PythonTestError` which can be casted from `DeserializationError`. WDYT?
</blockquote></details>

good, but the name is too general. rename the variant to `DeserializationError` or something

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 3 unresolved discussions (waiting on @dorimedini-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 62 at r5](https://reviewable.io/reviews/starkware-libs/committer/69#-Nwj6YQc6gBjuVBFYXpc:-Nwj7zBW4xGgzVujvz7a:b-3dql03) ([raw file](https://github.com/starkware-libs/committer/blob/db15b6d5e6f47df044d8e83d1562d388f488dfb4/crates/committer_cli/src/tests/python_tests.rs#L62)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

no more need for the `match`; the `?` operator automatically does `PythonTestError::from`
</blockquote></details>

nice :)
___
*[`crates/committer/src/deserialization/read.rs` line 25 at r1](https://reviewable.io/reviews/starkware-libs/committer/69#-NwiBrjn3miC-ezqfuCo:-Nwj7R7EDSCiqIZIHyYx:b-ep7asj) ([raw file](https://github.com/starkware-libs/committer/blob/f7973b113838b021c2a1a986587b5505e851f378/crates/committer/src/deserialization/read.rs#L25)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

good, but the name is too general. rename the variant to `DeserializationError` or something
</blockquote></details>

renamed to `DeserializationTestFailure`

**Review by dorimedini-starkware — COMMENTED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 1 unresolved discussion (waiting on @nimrod-starkware)

**Review by dorimedini-starkware — COMMENTED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 1 unresolved discussion (waiting on @nimrod-starkware)

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 1 unresolved discussion (waiting on @nimrod-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 112 at r8](https://reviewable.io/reviews/starkware-libs/committer/69#-NwnCJV21CszGNeiK_KJ:-NwnCJV21CszGNeiK_KK:bxtoouy) ([raw file](https://github.com/starkware-libs/committer/blob/c02f21565dcff44408d162cb9ffcb9b1bd03e8ef/crates/committer_cli/src/tests/python_tests.rs#L112)):*
> ```XML
>         res.push(a ^ b)
>     }
>     res
> ```

try `map`



_Suggestion:_
```XML
    x.iter().zip(y.iter()).map(|a, b| a ^ b).collect()
```

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 2 unresolved discussions (waiting on @nimrod-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 208 at r9](https://reviewable.io/reviews/starkware-libs/committer/69#-NwnlTdxCn4Di5Ioslnv:-NwnlTdxCn4Di5Ioslnw:b-4d0llm) ([raw file](https://github.com/starkware-libs/committer/blob/60e941f3cc392d3812f666567ec7e73aa257f6bb/crates/committer_cli/src/tests/python_tests.rs#L208)):*
> ```XML
> }
> 
> fn hash_state_diff(state_diff: &StateDiff) -> (Vec<u8>, Vec<u8>) {
> ```

add a docstring



_Code quote:_
```XML
fn hash_state_diff(state_diff: &StateDiff) -> (Vec<u8>, Vec<u8>) {
```
___
*[`crates/committer_cli/src/tests/python_tests.rs` line 319 at r9](https://reviewable.io/reviews/starkware-libs/committer/69#-Nwnj_136tLzfU59Hl-2:-Nwnj_136tLzfU59Hl-3:bk2faq) ([raw file](https://github.com/starkware-libs/committer/blob/60e941f3cc392d3812f666567ec7e73aa257f6bb/crates/committer_cli/src/tests/python_tests.rs#L319)):*
> ```XML
>     }
>     (keys_hash, values_hash)
> }
> ```

use macro to reduce boilerplate.
see if you can reduce boilerplate of other functions (I don't have ideas ATM)



_Suggestion:_
```XML
macro_rules! generate_storage_map_xor_hasher {
    ($fn_name:ident, $key_type:ty, $val_type:ty) => {
        fn $fn_name(inner_map: &HashMap<$key_type, $val_type>) -> (Vec<u8>, Vec<u8>) {
            let mut keys_hash = vec![0; 32];
            let mut values_hash = vec![0; 32];
            for (key, value) in inner_map {
                keys_hash = xor_hash(&keys_hash, &key.0.to_bytes_be());
                values_hash = xor_hash(&values_hash, &value.0.to_bytes_be());
            }
            (keys_hash, values_hash)
        }
    }
}

generate_storage_map_xor_hasher!(hash_storage_updates_map, StarknetStorageKey, StarknetStorageValue);
generate_storage_map_xor_hasher!(hash_class_hash_to_compiled_class_hash, ClassHash, CompiledClassHash);
generate_storage_map_xor_hasher!(hash_address_to_class_hash, ContractAddress, ClassHash);
generate_storage_map_xor_hasher!(hash_address_to_nonce, ContractAddress, Nonce);
```

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: 4 of 5 files reviewed, 2 unresolved discussions (waiting on @dorimedini-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 208 at r9](https://reviewable.io/reviews/starkware-libs/committer/69#-NwnlTdxCn4Di5Ioslnv:-Nwnr2XQCKEHfRy3X-9i:b-896fix) ([raw file](https://github.com/starkware-libs/committer/blob/60e941f3cc392d3812f666567ec7e73aa257f6bb/crates/committer_cli/src/tests/python_tests.rs#L208)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

add a docstring
</blockquote></details>

Done.
___
*[`crates/committer_cli/src/tests/python_tests.rs` line 319 at r9](https://reviewable.io/reviews/starkware-libs/committer/69#-Nwnj_136tLzfU59Hl-2:-Nwnr4PqERZDCPhl07z8:b-896fix) ([raw file](https://github.com/starkware-libs/committer/blob/60e941f3cc392d3812f666567ec7e73aa257f6bb/crates/committer_cli/src/tests/python_tests.rs#L319)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

use macro to reduce boilerplate.
see if you can reduce boilerplate of other functions (I don't have ideas ATM)
</blockquote></details>

Done.

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 1 unresolved discussion (waiting on @nimrod-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 316 at r10](https://reviewable.io/reviews/starkware-libs/committer/69#-NwoLWyA22vvO4iJzupy:-NwoLWyA22vvO4iJzupz:bla87ox) ([raw file](https://github.com/starkware-libs/committer/blob/a9150a22b54b9812a1fad96e97a3a29572d1008e/crates/committer_cli/src/tests/python_tests.rs#L316)):*
> ```XML
>     }
>     (keys_hash, values_hash)
> }
> ```

delete this; it is equivalent to `hash_storage_updates_map`

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69)* status: all files reviewed, 1 unresolved discussion (waiting on @dorimedini-starkware)

___
*[`crates/committer_cli/src/tests/python_tests.rs` line 316 at r10](https://reviewable.io/reviews/starkware-libs/committer/69#-NwoLWyA22vvO4iJzupy:-NwoNTjJCwyUC9WUrqgh:bxj2uy6) ([raw file](https://github.com/starkware-libs/committer/blob/a9150a22b54b9812a1fad96e97a3a29572d1008e/crates/committer_cli/src/tests/python_tests.rs#L316)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

delete this; it is equivalent to `hash_storage_updates_map`
</blockquote></details>

it's not actually.
`hash_storage_updates_map` uses `StarknetStorageKey` and `StarknetStorageValue` that wrap a felt. 
`hash_storage` uses `StorageKey` and `StorageValue` that wrap a `Vec<u8>`.

**Review by dorimedini-starkware — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/69#-:-NwoXTpH2VuKpDFO8mXS:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)
